### PR TITLE
prevent misuse of PyTuple_Pack

### PR DESCRIPTION
**Pull request by astrelsky**
This prevents simple mistakes such as the following

https://github.com/jpype-project/jpype/blob/66f8c6c07e051eb430326418798736004b0f833b/native%2Fpython%2Fpyjp_number.cpp#L390

### CODE REVIEW

**Comment by Thrameos** [comment]
Minor suggestion.... given that all uses of this function should be placed in a JPPyObject wrapper via ``call`` to we don't leak it would seem like combining the two would make sense.   It would take some minor reworking of a few sections which were already dangerous for exceptions.

**Comment by astrelsky** [comment]
> Minor suggestion.... given that all uses of this function should be placed in a JPPyObject wrapper via ``call`` to we don't leak it would seem like combining the two would make sense.   It would take some minor reworking of a few sections which were already dangerous for exceptions.

Agreed. I'll do this tonight.

File: native/python/pyjp_number.cpp [hunk 1]
```diff
@@ -404,8 +399,8 @@ JPPyObject PyJPNumber_create(JPJavaFrame &frame, JPPyObject& wrapper, const JPVa
 		jlong l = 0;
 		if (value.getValue().l != nullptr)
 			l = frame.CallBooleanMethodA(value.getJavaObject(), context->_java_lang_Boolean->m_BooleanValueID, nullptr);
-		PyObject *args = PyTuple_Pack(1, PyLong_FromLongLong(l));
-		return JPPyObject::call(PyLong_Type.tp_new((PyTypeObject*) wrapper.get(), args, nullptr));
```
Comment (astrelsky) [inline native/python/pyjp_number.cpp:408]: Was this and the two below a leak?

**Comment by astrelsky** [comment]
The changes were mostly painless. I only had to think about one change which was in `PyJPClass_FromSpecWithBases`. I left a comment where it should not be used in that function.

File: native/python/pyjp_number.cpp (hunk 1 above)
Comment (Thrameos) [reply native/python/pyjp_number.cpp:408]: Seems likely.  Once had leaks everwhere.   Hence need to write the C++ wrapper for python API.  Clearly this was missed.

**Review by Thrameos — APPROVED** [review]

**Comment by Thrameos** [comment]
Merge for upcoming release.

**Pull request merged**